analyzers/src/SonarAnalyzer.CSharp/Rules/LoggerMembersNamesShouldComply.cs

```diff
@@ -51,20 +63,31 @@ public sealed class LoggerMembersNamesShouldComply : ParametrizedDiagnosticAnaly
         KnownType.Castle_Core_Logging_ILogger);
 
     protected override void Initialize(SonarParametrizedAnalysisContext context) =>
-        context.RegisterNodeAction(c =>
+        context.RegisterCompilationStartAction(cc =>
         {
-            foreach (var memberData in Declarations(c.Node))
+            // TODO: Check for reference assemblies and exit-early.
```

Comment:
You can rebase on the other PR and remove the ToDo. Up to you.

Reply:
That's my plan, I left the TODO to break the QG and not merge this until we merge the helper PR and I can use the `StartCompilation` check :)